### DISCUSSION TITLE
How to verify the security of CLI tools rewritten in rust

### DISCUSSION THREAD

**Discussion by Bin-ly** [uutils/coreutils#5571] [Q&A] [answered]
Hello everyone, I have been researching some rust rewriting projects recently, would it be convenient to know your rewriting methods? Have you used tools like C2rust to help with rewriting? Also, how do you verify that CLI tools rewritten in rust are more secure than those written in C?

**Comment by sylvestre** [comment, marked as answer]
please use Discussions instead of issues for questions.

We didn't use C2rust because:
* License 
* For many contributors, the goal is to learn Rust on actual useful programs
* I don't know C2rust but I don't think it produces idiomatic Rust code


Rust removing some classes of issues, we are less worried about memory mgmt issues.
Besides that, we are fuzzing our code to find issues:
https://introspector.oss-fuzz.com/projects-overview
It didn't find any security issues.

**Reply by Bin-ly** [reply]
I also used fuzz to test the rust code I wrote, but it seemed that it could only verify whether there was a problem with my code. But there seems to be no way to prove that the code rewritten in rust is better than C. Do you know about the tools or methods to verify that the performance of rust code is better than C?

**Reply by sylvestre** [reply]
hyperfine is a great benchmarking tool.

**Reply by Bin-ly** [reply]
Thank you very much, I will go back and study this tool
